drivers/platform/stm32/stm32_i2c.c

```diff
@@ -43,6 +43,52 @@
 #include "no_os_i2c.h"
 #include "stm32_i2c.h"
 
+/**
+ * @brief i2c_table contains the pointers towards the I2C buses
+*/
+static void *i2c_table[4];
+
+/**
+ * @brief STM32 I2C bus initializer
+*/
+int32_t stm32_i2c_initialize_bus(uint32_t bus_number,
+				 struct no_os_i2c_init_param *param)
+{
+	i2c_table[bus_number] = (struct no_os_i2cbus_desc *)no_os_calloc(1,
+				sizeof(*no_os_i2cbus_desc));
+
+	if (!i2c_table[bus_number])
+		return -ENOMEM;
+
+	no_os_mutex_init(i2c_table[bus_number].mutex);
+
+	if (i2c_table[bus_number].mutex == NULL)
```

Comment:
i'd keep the code consistent: here you to `if (x == NULL)` and right above you do `if (!x)`